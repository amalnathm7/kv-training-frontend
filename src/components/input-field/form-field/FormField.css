.form-item {
    width: 30%;
    padding: 20px;
}

.form-field {
    margin-top: 15px;
    align-items: center;
    justify-content: center;
    height: 55px;
    width: 100%;
    border-radius: 5px;
    caret-color: #03AEEE;
    border: 2px solid lightgray;
    padding: 10px;
    outline: 0;
    background: #FAFAFA 0% 0% no-repeat padding-box;
    border: 1px solid #D9D9D9;
}

.form-label {
    margin-top: 20px;
    font-size: 14px;
    color: #555555;
    font-weight: bold;
}

.form-error {
    margin-top: 10px;
    text-indent: 10px;
    font-size: 14px;
    color: red;
}

.spacing {
    height: 16px;
}