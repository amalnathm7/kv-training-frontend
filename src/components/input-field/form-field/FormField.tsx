import React from "react";
import "./FormField.css";

export type FormFieldPropsType = {
    value: string | number | string[],
    onChange: (event) => void,
    label: string,
    placeholder: string,
    type: string,
    showError: boolean,
    disabled: boolean
}

const FormField: React.FC<FormFieldPropsType> = (props) => {
    return (
        <div className="form-item">
            {props.label.length === 0 && <div className="spacing"></div>}
            <label className="form-label">{props.label}</label>
            <input className="form-field" placeholder={props.placeholder} onChange={props.onChange} type={props.type} value={props.value} disabled={props.disabled}></input>
            {props.showError && <label className="form-error">Enter {props.label.trim().length !== 0 ? props.label.toLowerCase() : props.placeholder.toLowerCase()}</label>}
        </div>
    );
};

export default FormField;
