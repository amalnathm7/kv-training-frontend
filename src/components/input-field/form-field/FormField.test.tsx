import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import FormField, { FormFieldPropsType } from './FormField';

describe('FormField Props Test', () => {
  test('If FormField rendered correctly', () => {
    const props: FormFieldPropsType = {
      value: 'FormField-value',
      onChange: () => {},
      label: 'FormField-label',
      placeholder: 'FormField-placeholder',
      showError: false,
      disabled: false,
      type: 'text'
    };

    const element = render(<FormField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If FormField rendered correctly with error', () => {
    const props: FormFieldPropsType = {
      value: 'FormField-value',
      onChange: () => {},
      label: 'FormField-label',
      placeholder: 'FormField-placeholder',
      showError: true,
      disabled: false,
      type: 'text'
    };

    const element = render(<FormField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If FormField rendered correctly with no label', () => {
    const props: FormFieldPropsType = {
      value: 'FormField-value',
      onChange: () => {},
      label: '',
      placeholder: 'FormField-placeholder',
      showError: false,
      disabled: false,
      type: 'text'
    };

    const element = render(<FormField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If FormField rendered correctly with error and no label', () => {
    const props: FormFieldPropsType = {
      value: 'FormField-value',
      onChange: () => {},
      label: '',
      placeholder: 'FormField-placeholder',
      showError: true,
      disabled: false,
      type: 'text'
    };

    const element = render(<FormField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If disabled FormField rendered correctly', () => {
    const props: FormFieldPropsType = {
      value: 'FormField-value',
      onChange: () => {},
      label: '',
      placeholder: 'FormField-placeholder',
      showError: false,
      disabled: true,
      type: 'text'
    };

    const element = render(<FormField {...props} />);

    expect(element).toMatchSnapshot();
  });
});
