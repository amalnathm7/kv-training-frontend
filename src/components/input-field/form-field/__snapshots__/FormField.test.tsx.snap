// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField Props Test If FormField rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-item"
      >
        <label
          class="form-label"
        >
          FormField-label
        </label>
        <input
          class="form-field"
          placeholder="FormField-placeholder"
          type="text"
          value="FormField-value"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-item"
    >
      <label
        class="form-label"
      >
        FormField-label
      </label>
      <input
        class="form-field"
        placeholder="FormField-placeholder"
        type="text"
        value="FormField-value"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormField Props Test If FormField rendered correctly with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-item"
      >
        <label
          class="form-label"
        >
          FormField-label
        </label>
        <input
          class="form-field"
          placeholder="FormField-placeholder"
          type="text"
          value="FormField-value"
        />
        <label
          class="form-error"
        >
          Enter 
          formfield-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-item"
    >
      <label
        class="form-label"
      >
        FormField-label
      </label>
      <input
        class="form-field"
        placeholder="FormField-placeholder"
        type="text"
        value="FormField-value"
      />
      <label
        class="form-error"
      >
        Enter 
        formfield-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormField Props Test If FormField rendered correctly with error and no label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-item"
      >
        <div
          class="spacing"
        />
        <label
          class="form-label"
        />
        <input
          class="form-field"
          placeholder="FormField-placeholder"
          type="text"
          value="FormField-value"
        />
        <label
          class="form-error"
        >
          Enter 
          formfield-placeholder
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-item"
    >
      <div
        class="spacing"
      />
      <label
        class="form-label"
      />
      <input
        class="form-field"
        placeholder="FormField-placeholder"
        type="text"
        value="FormField-value"
      />
      <label
        class="form-error"
      >
        Enter 
        formfield-placeholder
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormField Props Test If FormField rendered correctly with no label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-item"
      >
        <div
          class="spacing"
        />
        <label
          class="form-label"
        />
        <input
          class="form-field"
          placeholder="FormField-placeholder"
          type="text"
          value="FormField-value"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-item"
    >
      <div
        class="spacing"
      />
      <label
        class="form-label"
      />
      <input
        class="form-field"
        placeholder="FormField-placeholder"
        type="text"
        value="FormField-value"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
